app/javascript/src/utils/compiler.js

```diff
@@ -412,6 +412,17 @@ function evaluateForLoops(joinedItems) {
   return joinedItems
 }
 
+const openToClosingBracketsMap = {
+  "(": ")",
+  "[": "]"
+}
+const openBracketRegex = new RegExp(
+  Object.keys(openToClosingBracketsMap)
+    .map((c) => `\\${ c }`)
+    .join("|"),
+  "g"
+)
+
```

Comment:
Let's put consts like this either in the function they belong in or in a separate file, otherwise they might get lost

Reply:
I didn't want to put them inside the function since it could be called more than once per `@each` match, and I didn't think they'd be good in a separate file since they are specific to `parseArrayValues()` (e.g. there is no `"{": "}"` entry)

Reply:
I see, let's put them at the top of the file then

Reply:
How about having this file structure:
```js
./compiler
├── array.js
│   ├── const openToClosingBracketsMap
│   ├── const unescapedOpenBracketRegex
│   ├── const escapedCharRegex
│   └── parseArrayValues(input)
├── expression.js
│   ├── const comparisonOperators
│   ├── const sortedComparisonOperatorsSymbols
│   ├── getExpressionTree(expression)
│   └── evaluateExpressionTree(tree)
└── index.js // before: compiler.js
```

only issue is where to put shared functions like `removeSurroundingParenthesis(source)` (a shared.js?)

Reply:
How about separating it even more?
- Compiler
  - Mixin.js
  - For.js
  - Each.js
  .....
  - helpers.js

Reply:
Let's discuss it later some more, not in this PR